pylint/checkers/refactoring.py

```diff
@@ -527,6 +533,29 @@ def visit_assign(self, node):
 
     visit_return = visit_assign
 
+    def _check_consider_using_join(self, aug_assign):
+        """
+        We start with the augmented assignment and work our way upwards.
+        Names of variables for nodes if match successful:
+        for number in ['1', '2', '3']  # for_loop
+            result += number  # aug_assign
+        """
+        for_loop = aug_assign.parent
+        if not isinstance(for_loop, astroid.node_classes.For):
+            return
+
+        is_concat_loop = (aug_assign.op == '+='
```

Comment:
I think I prefer the former approach, in which we were checking the type of the initial assignment, not the type of the augassign node.